src/ert/config/response_config.py

```diff
@@ -5,14 +5,17 @@
 import xarray as xr
 
 from ert.config.parameter_config import CustomDict
+from ert.substitution_list import SubstitutionList
 
 
 @dataclasses.dataclass
 class ResponseConfig(ABC):
     name: str
 
     @abstractmethod
-    def read_from_file(self, run_path: str, iens: int) -> xr.Dataset: ...
+    def read_from_file(
+        self, run_args: str, iens: int, itr: int, substitution_list: SubstitutionList
```

Comment:
Could we perhaps use just a dict[str,str] instead of `SubstitutionList`, makes the API less dependent on ert internals, or at least make the argument optional with a default of an empty `SubstitutionList`.

Reply:
Yea, I think it would be nice if we had `substitution_list: dict[str, str]` in `ert_config.py`. Don't really think subtyping dict is the way to go here, we could just have substitution functions that do the work.

Reply:
> Could we perhaps use just a dict[str,str] instead of SubstitutionList, makes the API less dependent on ert internals, or at least make the argument optional with a default of an empty SubstitutionList.

It might be a good idea to just make the argument optional. 

> we could just have substitution functions that do the work.

Sure but this is a refactor that I don't think is needed to solve this issue.